src/renderer/modals/FullNode/steps/StepConnectDevice.js

```diff
@@ -0,0 +1,151 @@
+  const [device, setDevice] = useState(null);
+  const [scanStatus, setScanStatus] = useState<ConnectionStatus>(connectionStatus.IDLE);
+
+  useEffect(() => {
+    if (device) {
+      const currency = getCryptoCurrencyById("bitcoin");
+      command("scanDescriptors")({
+        deviceId: device.deviceId,
+        currency,
+        limit: 10,
+      })
+        // $FlowFixMe I don't know what you want.
+        .pipe(reduce((acc, item) => acc.concat({ descriptor: item }), []))
+        .subscribe(descriptors => {
+          setScanStatus(connectionStatus.SUCCESS);
+          setScannedDescriptors(descriptors);
+        });
+    }
+  }, [device, setScannedDescriptors]);
+
+  return (
+    <>
+      <TrackPage category="FullNode" name="Step3" currencyName={currency.name} />
+      {scanStatus === connectionStatus.IDLE ? (
+        <DeviceAction
+          action={action}
+          request={{ currency }}
+          onResult={({ device }) => {
+            setDevice(device);
+            setScanStatus(connectionStatus.PENDING);
+          }}
+        />
+      ) : scanStatus === connectionStatus.SUCCESS ? (
+        <Box alignItems="center">
+          <CheckWrapper size={50}>
+            <IconCheck size={20} />
+          </CheckWrapper>
+          <Text
+            ff="Inter|SemiBold"
+            textAlign={"center"}
+            mt={32}
+            fontSize={6}
+            color="palette.text.shade100"
+          >
+            <Trans i18nKey="fullNode.modal.steps.device.connectionSteps.success.header" />
+          </Text>
+          <Text
+            ff="Inter|Regular"
+            mt={2}
+            textAlign={"center"}
+            fontSize={3}
+            color="palette.text.shade50"
+          >
+            <Trans i18nKey="fullNode.modal.steps.device.connectionSteps.success.description" />
+          </Text>
+        </Box>
+      ) : (
+        <Box alignItems="center">
+          <BigSpinner size={50} />
+          <Text
+            ff="Inter|SemiBold"
+            textAlign={"center"}
+            mt={32}
+            fontSize={6}
+            color="palette.text.shade100"
+          >
+            <Trans i18nKey="fullNode.modal.steps.device.connectionSteps.pending.header" />
+          </Text>
+          <Text
+            ff="Inter|Regular"
+            mt={2}
+            textAlign={"center"}
+            fontSize={3}
+            color="palette.text.shade50"
+          >
+            <Trans i18nKey="fullNode.modal.steps.device.connectionSteps.pending.description" />
+          </Text>
+        </Box>
+      )}
+    </>
+  );
+};
+
+export const StepDeviceFooter = ({
+  onClose,
+  onStepChange,
+  scannedDescriptors,
+}: {
+  onClose: () => void,
+  onStepChange: FullNodeSteps => void,
+  scannedDescriptors: any,
+}) => {
+  const onOpenUserDataFolder = useCallback(() => {
+    const userDataDirectory = resolveUserDataDirectory();
+    shell.showItemInFolder(userDataDirectory);
+  }, []);
```

Comment:
why don't we have an existing component that do exactly that? It seems we repeat ourself with the same button that exists in settings?

Reply:
We do, had to tweak it to be generic, but we do now.